polling_stations/apps/data_importers/data_types.py

```diff
@@ -0,0 +1,318 @@
+            else:
+                return e.council.council_id
+
+    def update_uprn_to_council_model(self, polling_station_lookup=None):
+        if not polling_station_lookup:
+            polling_station_lookup = self.get_polling_station_lookup()
+
+        uprns_in_council = UprnToCouncil.objects.filter(lad=self.council_id)
+        for polling_station_id, uprns in polling_station_lookup.items():
+            uprns_in_council.filter(uprn__in=uprns).update(
+                polling_station_id=polling_station_id
+            )
+
+
+class DistrictSet(CustomSet, AssignPollingStationsMixin):
+    def build_namedtuple(self, element):
+
+        # MultiPolygon is mutable, so we must serialize it to store in a tuple
+        area = element["area"].ewkb  # use ewkb so it encodes srid
+
+        return District(
+            element.get("name", ""),
+            element["council"],
+            element["internal_council_id"],
+            element.get("extra_id", ""),
+            area,
+            element.get("polling_station_id", ""),
+        )
+
+    def save(self):
+        districts_db = []
+        for district in self.elements:
+            record = PollingDistrict(
+                name=district.name,
+                council=district.council,
+                internal_council_id=district.internal_council_id,
+                extra_id=district.extra_id,
+                area=district.area,
+                polling_station_id=district.polling_station_id,
+            )
+            districts_db.append(record)
+        PollingDistrict.objects.bulk_create(districts_db)
+        self.saved = True
+
+    def get_uprns_by_district(self, cursor):
+        cursor.execute(
+            """
+                SELECT a.uprn, d.polling_station_id
+                FROM addressbase_address a
+                    JOIN addressbase_uprntocouncil u
+                    ON a.uprn = u.uprn
+                    JOIN pollingstations_pollingdistrict d
+                    ON ST_Contains(d.area, a.location)
+                WHERE d.council_id=%s
+                AND u.lad=%s
+            """,
+            [self.council_id, self.council_id],
+        )
+        return cursor
+
+    def get_uprns_by_district_join_stations(self, cursor):
+        cursor.execute(
+            """
+                SELECT a.uprn, s.internal_council_id
+                FROM addressbase_address a
+                    JOIN addressbase_uprntocouncil u
+                    ON a.uprn = u.uprn
+                    JOIN pollingstations_pollingdistrict d
+                    ON ST_Contains(d.area, a.location)
+                    JOIN pollingstations_pollingstation s
+                    ON s.polling_district_id = d.internal_council_id
+                WHERE d.council_id=%s
+                AND u.lad=%s
+                AND s.council_id=%s
+            """,
+            [self.council_id, self.council_id, self.council_id],
+        )
+        return cursor
```

Comment:
I think it would be neater here to make these two functions

```py
def get_uprns_by_district(self):
    cursor = connection.cursor()
    cursor.execute('big SQL statement')
    return cursor.fetchall()

def get_uprns_by_district_join_stations(self):
    cursor = connection.cursor()
    cursor.execute('big SQL statement')
    return cursor.fetchall()
```

rather than passing a `cursor` around.